File: Exemplos/App_eXperience_Flutter/Elgin_AppExperience_Flutter-font/android/app/src/main/java/com/elgin/flutter_m8/TefEnums/FormaPagamento.java
package com.elgin.flutter_m8.TefEnums;

import java.text.Normalizer;

public enum FormaPagamento {
    CREDITO("credito"), DEBITO("debito"), TODOS("todos");

    //String rotulo utilizado para aferir os parâmetros enviados a partir da plataforma Flutter.
    private final String rotulo;

    FormaPagamento(String rotulo) {
        this.rotulo = rotulo;
    }

    public static FormaPagamento fromRotulo(String rotulo) {
        for (FormaPagamento formaPagamento : values()) {
            if (formaPagamento.rotulo.equals(rotulo)) {
                return formaPagamento;
            }
        }
        throw new IllegalArgumentException("Nenhuma forma de pagamento válida enviada!");
    }
}

File: Exemplos/App_eXperience_Flutter/Elgin_AppExperience_Flutter-font/lib/pages/tef/FormaPagamento.dart
part of './TefPage.dart';

//Formas de pagamento disponíveis.
class FormaPagamento {
  static const FormaPagamento CREDITO = FormaPagamento._('credito');
  static const FormaPagamento DEBITO = FormaPagamento._('debito');
  static const FormaPagamento TODOS = FormaPagamento._('todos');

  final String _rotulo;

  const FormaPagamento._(this._rotulo);

  @override
  toString() => this._rotulo;
}